src/net/ssl-context.cpp

```diff
@@ -75,7 +80,7 @@ SslContext::SslContext(std::string ca_bundle_path) {
 }
 
 SslContext::~SslContext() {
-    EVP_cleanup();
+    //EVP_cleanup();  // XXX Should this be called on every destruction?
```

Comment:
So if we make this into a singleton the call to `OpenSSL_add_all_algorithms` will happen only once and if this does get called on every destruction it will mean that they will be unloaded from the table, so I guess it's good to remove it.

Moreover since EVP_cleanup() has no more effect since OpenSSL 1.1.0 we can I think not worry too much about the extra memory usage for older versions.